### ISSUE TITLE
auth state in different clients for SDK and react is not synced

### ISSUE THREAD

**Issue by chrtze** [nhost/nhost#270]
I am using the newly released hooks for authentication within nextjs with the client from `@nhost/client`. In my app, I need to fetch user data stored in a custom table after authentication. I tried using the apollo `useQuery` together with `@nhost/react-apollo` and `nhost.graphql.request` from `@nhost/nhost-js`. Both functions did not include the authorization header so that the data could not be fetched.

It makes sense that the request from `nhost-js` doesn't work, as the client probably doesn't know about the auth state of the react client (though it would be cool if it would be the same client for react and the vanilla sdk).

It seems like using the apollo client would be more suitable in my scenario, but it seems like the ApolloProvider doesn't notice the auth state as well. Currently I am using this workaround:

```
const isAuthenticated = useAuthenticated();
const accessToken = useAccessToken();

  useEffect(() => {
    if (isAuthenticated && accessToken) {
      const getData = async () => {
        // @todo should not include the accessToken here
        // https://docs.nhost.io/platform/database/permissions
        // due to hasura permissions, this will only return data belonging to the user
        const data = await nhost.graphql.request(GET_DATA, null, {
          headers: { Authorization: `Bearer ${accessToken}` },
        });

        setData(data);
      };
      getData();
    } else {
      setData(null);
    }
  }, [isAuthenticated, accessToken]);

**Comment by plmercereau** [comment]
Hello Christopher, and thanks for this issue.
To be able to use `@nhost/nhost-js` inside the above code totally makes sense, and [we are currently working on it](https://github.com/nhost/nhost/pulls). It will be hopefully available soon.
In the meantime, I'm curious about how you set up the Apollo provider, as it is supposed to work already, I'll be happy to check your code

**Comment by plmercereau** [comment]
Hello again,
This is solved in [`@nhost/react@0.3.0`](https://github.com/nhost/nhost/releases/tag/%40nhost%2Freact%400.3.0).
Don't hesitate to answer to this issue if it didn't solve your problem, and I'll reopen it

**Comment by chrtze** [comment]
This works perfectly. Thanks!
